src/components/common/RestEntity/README.md

```diff
@@ -0,0 +1,882 @@
+
+- Course
+
+- Module
+
+Each one of those will probably be presented either as a one-off view or a list of multiple. Say that a user click on a Profile Page — now my component would have to fetch data for a **SINGLE USER**. And then that user clicks on their `ManageUsers` dashboard to view all of the users in their system; now `ManageUsers` would have to fetch data for a **LIST OF USERS**.
+
+You can probably pretty easily imagine a situation for a Single view vs a List view for each of the four big data types above.
+
+- User:
+  - Profile Page — Single User
+  - User Management Dash — List of Users
+- Program
+  - Edit Program Form — Fetch Single Program
+  - Program Dashboard — List of Programs
+- And so on, and so forth
+
+
+Handling all of that data fetching IN each component on top of managing application state, user-flow, interactivity, and everything else would result in some wildly complex components.
+
+But anything fetching data is going to do it in relatively the same way. And each component really just needs to be able to decide how to DISPLAY that information to the user. 
+
+Enter `RestEntity`. This component is all about data-fetching.
+
+---
+
+## RestEntity Overview
+
+The `RestEntity` component is a compound component that utilizes the React Context API to handle data-fetching and consume relational links. Though using such a component will feel new and different to many of you, I promise it appears more intimidating than it really is. Once you're working with it,
+it'll feel very similar to the same old React that you're used to.
+
+The whole purpose of the `RestEntity` is to simplify the process of interacting with HAL-shaped data from our RESTful, HATEOAS-oriented backend. That is to say, every single data object has hypermedia links. Instead of worrying about complex Redux thunks and/or over-the-top routing, our `RestEntity` is going to take whatever endpoint you want to feed it and get the data you care about. Then, it'll give control back to you so you can do what you're so good at doing: building components to PRESENT information.
+
+The `RestEntity` has a couple parts, most of which are completely optional.
+
+```javascript
+const EndpointComponent = props => {
+  return (
+    
+    <RestEntity href=“/endpoint”>
+
+      {/* Successful Fetch State */}  
+      <RestEntity.Singleton 
+        component={data => (
+          <SomeComponent {...data} />
+        )}
+      />
+
+      {/* Error State */}  
+      <RestEntity.Error>
+        <div>Oh no...! An error has occurred!</div>
+      </RestEntity.Error>
+
+      {/* Loading State */}  
+      <RestEntity.Loading>
+        <Spinner />
+      </RestEntity.Loading>
+
+    </RestEntity>
+  );
+
+};
+
+```
+
+Quick note on this "dot-notation" with components, for anyone who hasn't ever encountered that before: React Components are JavaScript functions (even the class components). And JavaScript functions are JavaScript OBJECTS. And JS Objects have PROPERTIES.
```

Comment:
This might be a little too in depth, you can also just link to a related article that explains these concepts as they are pretty fundamental.

Reply:
fair. I just didn’t know the best way to explain the compound component part for anyone who hasn’t seen that pattern before.  

but i will strip some of that out and link.